tee-worker/app-libs/parentchain-interface/src/integritee/event_handler.rs

```diff
@@ -201,17 +222,158 @@ impl ParentchainEventHandler {
+		let account_ids: Vec<AccountId> =
+			scores_storage_keys.iter().filter_map(key_to_account_id).collect();
+
+		let delegator_state_storage_keys: Vec<Vec<u8>> = account_ids
+			.iter()
+			.map(|account_id| {
+				storage_map_key(
+					"ParachainStaking",
+					"DelegatorState",
+					account_id,
+					&StorageHasher::Blake2_128Concat,
+				)
+			})
+			.collect();
+		let delegator_states: BTreeMap<AccountId, Delegator<AccountId, Balance>> = self
+			.ocall_api
+			.get_multiple_storages_verified(
+				delegator_state_storage_keys,
+				&block_header,
+				&ParentchainId::Litentry,
+			)
+			.map_err(|_| Error::Other("Failed to get multiple storages".into()))?
+			.into_iter()
+			.filter_map(|entry| {
+				let storage_key = decode_storage_key(entry.key)?;
+				let account_id = key_to_account_id(&storage_key)?;
+				let delegator = entry.value?;
+				Some((account_id, delegator))
+			})
+			.collect();
+
+		let id_graphs_storage_keys: Vec<Vec<u8>> = account_ids
+			.iter()
+			.map(|account_id| {
+				storage_map_key(
+					"IdentityManagement",
+					"IDGraphs",
+					&Identity::from(account_id.clone()),
+					&StorageHasher::Blake2_128Concat,
+				)
+			})
+			.collect();
+
+		let shard = executor.get_default_shard();
+
+		let accounts_graphs = self
+			.state_handler
+			.execute_on_current(&shard, |state, _| {
+				let mut id_graphs_accounts: BTreeMap<AccountId, Vec<AccountId>> = BTreeMap::new();
+				for id_graph_storage_key in id_graphs_storage_keys.iter() {
+					let id_graph: Vec<(Identity, IdentityContext<Runtime>)> = state
+						.iter_prefix::<Identity, IdentityContext<Runtime>>(id_graph_storage_key)
+						.unwrap_or_default();
+					let graph_accounts: Vec<AccountId> = id_graph
+						.iter()
+						.filter_map(|(identity, _)| identity.to_account_id())
+						.collect();
+					if let Some(account_id) = key_to_account_id(id_graph_storage_key) {
+						id_graphs_accounts.insert(account_id, graph_accounts);
+					}
+				}
+
+				id_graphs_accounts
+			})
+			.map_err(|_| Error::Other("Failed to get id graphs".into()))?;
+
+		let extrinsic_sender = ParachainExtrinsicSender::new();
+		let update_token_staking_amount_call_index = self
+			.node_metadata_repository
+			.get_from_metadata(|m| m.update_token_staking_amount_call_indexes())
+			.map_err(|_| {
+				Error::Other(
+					"Metadata retrieval for update_token_staking_amount_call_indexes failed".into(),
+				)
+			})?
+			.map_err(|_| Error::Other("Invalid metadata".into()))?;
+
+		for account_id in account_ids.iter() {
+			let default_id_graph = Vec::new();
+			let id_graph = accounts_graphs.get(account_id).unwrap_or(&default_id_graph);
+			let staking_amount: Balance = id_graph
+				.iter()
+				.filter_map(|identity| {
+					let delegator = delegator_states.get(identity)?;
+					Some(delegator.total)
+				})
+				.sum();
+			let call = OpaqueCall::from_tuple(&(
+				update_token_staking_amount_call_index,
+				account_id,
+				staking_amount,
+			));
+			extrinsic_sender
```

Comment:
In multiworker setup this would lead to duplicate calls, have you considered executing this logic by sidechain block producer only ?

Reply:
This is a very good point, I have not considered it 👍🏼